via/static/scripts/video_player/components/VideoPlayerApp.tsx

```diff
@@ -47,6 +49,47 @@ function isScrollToRangeEvent(e: Event): e is ScrollToRangeEvent {
   );
 }
 
+type FilterInputProps = {
+  elementRef: Ref<HTMLElement | undefined>;
+  setFilter: (filter: string) => void;
+  filter: string;
+};
+
+function FilterInput({ elementRef, setFilter, filter }: FilterInputProps) {
```

Comment:
Certainly could extract this component entirely, but for expediency (mostly in tests) and dev context I've left it internal to `VideoPlayerApp` for the moment, especially since things are moving around so much right now.